openmsupply/application-manager-web-app
# #380 review page and page elements

Fixes: #380 
Fixes: #349

Branched from: #386

### Changes

** PageElements** 
* Added to isReview 
* Simplified JSX for other views
* Changed prop to take PageElement (need this for thisReviewLatestResponse etc..)
* Added ReviewResponseComponent (shows one line of review)
* Added ReviewBotton (the one to edit or review the review)
** SectionWrapper**
* Simplified JSX
* Added extraSectionTitleContent prop and extraPageContent (both are methods that will render something extra, take section or page as props)
** DecisionAreaNew **
* Copied old one and simplified
** Application Page **
* getCurrentElements returns PageElement[] vs ElementState[] to (to pass to PageElements component)
** useGetFullReviewStructure **
* add reviewInfo and status to structure
** getReviewQuery **
* also returns status
** ReviewPage **
* new page to display review
* passing some components to render extra info in SectionWrapper


### Testing

* use test branch #380-testing and back end branch #386frontend-helpers
* start a review for ABC123 (with reviewer 1 or 2, using existing UI)
* navigate to http://localhost:3000/applicationNew/ABC123/reviewTest (with either of the reviewer, make sure to update test page with 1020 -> reviewer 1 or 1021 reviewer 2)
* see extra bit of information that can be use by progress in the section header, it should change as you make changes
* note, when users change review assignment should change
